packages/syft/src/syft/types/twin_object.py

```diff
@@ -87,9 +87,11 @@ def _save_to_blob_storage(self) -> SyftError | None:
             self.syft_node_location,
             self.syft_client_verify_key,
         )
-        # self.mock_obj._set_obj_location_(
-        #     self.syft_node_location,
-        #     self.syft_client_verify_key,
-        # )
+        self.mock_obj._set_obj_location_(
+            self.syft_node_location,
+            self.syft_client_verify_key,
+        )
+        mock_store_res = self.mock_obj._save_to_blob_storage()
```

Comment:
We need to validate if the blob_storage_entry_id has read permissions for mock objects. 
https://github.com/OpenMined/PySyft/pull/8896/files#diff-4ef06b0973fb9fa22d186235a17d570c846c26cb8dc87ed184dd470f04f81fc6L123

here if during action.set, if its a TwinObject, and the `mock_obj` has blob_storage_entry_id then we should add READ permission to it.